elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -86,4 +87,16 @@ default void flush() {
         // default to ignoring criteria
         return loadObjects(entityClass);
     }
+
+    /**
+     * Filter a collection by the Predicates in filterScope.
+     *
+     * @param collection the collection to filter
+     * @param type
+     * @param filterScope the FilterScope containing the set of Predicates
+     * @return the filtered collection
+     */
+    default Collection filterCollection(Collection collection, String type, FilterScope<?> filterScope) {
```

Comment:
The default implementation should actually perform the filters on the collection by iterating through the elements and then evaluating the predicates.